pkg/controller/metrics/metrics.go

```diff
@@ -246,6 +302,48 @@ func (mc *Calculator) Start(ctx context.Context) error {
 						mcLog.Infof("cleared metric: %v", metricClusterDeploymentSyncsetPaused)
 					}
 				}
+
+				// Check if cluster is currently transitioning and set the appropriate metrics
+				poolNS := "unknown"
+				if cd.Spec.ClusterPoolRef != nil {
+					poolNS = cd.Spec.ClusterPoolRef.Namespace
+				}
+				hibernatingCond := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions,
+					hivev1.ClusterHibernatingCondition)
+				readyCond := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions,
+					hivev1.ClusterReadyCondition)
+				if readyCond.Reason == hivev1.ReadyReasonStoppingOrHibernating &&
+					hibernatingCond.Reason != hivev1.HibernatingReasonHibernating {
+					transitionTime := time.Since(readyCond.LastTransitionTime.Time).Seconds()
+					if ShouldLogDurationMetric(MetricStoppingClusters, transitionTime) {
```

Comment:
The func we call could incorporate both the conditional and the observe, and be called more simply like `MaybeObserveDurationMetric(MetricStoppingClusters, time.Since(...), cd)`. That'd depend on all such metrics having the same labels... but I say since they do right now we can simplify -- break it up later if necessary.

Reply:
...and factor out the bit that determines the list of label values

...and call :arrow_up: from a new function that encapsulates the `Delete()`s as well.